utils/build-script

```diff
@@ -934,7 +717,10 @@ class BuildScriptInvocation(object):
         # Build...
         for host_target in all_hosts:
             # FIXME: We should only compute these once.
-            config = HostSpecificConfiguration(host_target.name, self)
+            try:
+                config = HostSpecificConfiguration(self.args, host_target.name)
+            except argparse.ArgumentError as e:
+                exit_rejecting_arguments(e.message)
```

Comment:
Here and in the previous code change: `BuildScriptInvocation` was calling a function `exit_rejecting_arguments` in the middle of the calculation of `platforms_to_skip_test`. Since those variables are only used in `HostSpecificConfiguration`, I moved them there, and to reproduce the previous behaviour, if `HostSpecificConfiguration` finds some problem with the passed arguments, it will raise `ArgumentError`, which will be dealt here. That allows the users of `HostSpecificConfiguration` decide how to act in the case of an error.